tokio-stream/src/stream_ext/map_while.rs

```diff
@@ -0,0 +1,70 @@
+use crate::Stream;
+
+use core::fmt;
+use core::pin::Pin;
+use core::task::{Context, Poll};
+use pin_project_lite::pin_project;
+
+pin_project! {
+    /// Stream for the [`map_while`](super::StreamExt::map_while) method.
+    #[must_use = "streams do nothing unless polled"]
+    pub struct MapWhile<St, F> {
+        #[pin]
+        stream: Option<St>,
```

Comment:
I don't think this `Option` is needed. Users who need this behavior can use `.fuse()` explicitly.

See also implementation of [Iterator::map_while](https://doc.rust-lang.org/stable/src/core/iter/adapters/map_while.rs.html#43)

Reply:
That's a very good point. Will address

Reply:
There are a few combinators (`skip_while` and `take_while` at least) where we keep an extra piece of state (like the `Option` here) to remember whether we were done. I guess we don't need those either! They can be fixed in a different PR though